src/Uno.UI.Composition/Composition/ExpressionAnimation.cs

```diff
@@ -0,0 +1,46 @@
+#nullable enable
+
+using System;
+
+namespace Microsoft.UI.Composition;
+
+public partial class ExpressionAnimation : CompositionAnimation
+{
+	private AnimationExpressionSyntax? _parsedExpression;
+
+	internal ExpressionAnimation(Compositor compositor) : base(compositor)
+	{
+	}
+
+	public string? Expression { get; set; }
+
+	private protected override void OnPropertyChangedCore(string? propertyName, bool isSubPropertyChange)
+	{
+		if (_parsedExpression is not null)
+		{
+			RaisePropertyChanged();
+		}
+	}
+
+	internal override object? Start()
+	{
+		if (Expression is null)
+		{
+			throw new InvalidOperationException("Property 'Expression' should not be null when starting an ExpressionAnimation");
+		}
+
+		// TODO: Check what to do if this is a second Start call and we already have non-null _parsedExpression;
+		_parsedExpression = new ExpressionAnimationParser(Expression).Parse();
+
+		return _parsedExpression.Evaluate(this);
+	}
+
+	internal override object? Evaluate()
```

Comment:
Not from WinUI. But I also can't see how this could be implemented with generics. What type argument will the callers pass? It will be unknown in many cases I think.

I would guess boxing in this code path isn't going to be significant in most cases